pkg/sortition/internal/local.go

```diff
@@ -88,6 +88,28 @@ func (lc *localChain) IsOperatorInPool() (bool, error) {
 	return ok, nil
 }
 
+func (lc *localChain) IsOperatorUpToDate() (bool, error) {
+	lc.sortitionPoolMutex.RLock()
+	defer lc.sortitionPoolMutex.RUnlock()
+
+	lc.eligibleStakeMutex.RLock()
+	defer lc.eligibleStakeMutex.RUnlock()
+
+	stakingProvider, isRegistered := lc.operatorToStakingProvider[lc.operatorAddress]
+	if !isRegistered {
+		return false, errOperatorUnknown
+	}
+
+	eligibleStake, hasStake := lc.eligibleStake[stakingProvider]
+	weight, isInPool := lc.sortitionPool[lc.operatorAddress]
+
+	if !isInPool {
```

Comment:
Might be slightly easier to read, up to you..
```
        if isInPool {
		return weight.Cmp(eligibleStake) == 0, nil
	}

	return !hasStake || eligibleStake.Cmp(big.NewInt(0)) == 0, nil
```

Reply:
That's a good idea. I changed it in #3077 to avoid conflicts. See de8cd7075be3a3014829fcd744fbcad426ee0357 please.